### PR TITLE
Simplify if else block in GenerateConfig()

### PR DESCRIPTION
**Pull request by steiler**

### CODE REVIEW

**Comment by hellt** [comment]
@steiler would you mind rebasing this on top of main to get rid of the merge conflict that happened after suppress-config merge? Thanks!

**Comment by steiler** [comment]
Done. Please take another look if the logic still makes sense.
I'm positive, but please double-check.

File: nodes/srl/srl.go [hunk 1]
```diff
@@ -486,13 +485,12 @@ func (s *srl) createSRLFiles() error {
...
 
-		cfgTemplate := string(c)
-
-		err = s.GenerateConfig(dst, cfgTemplate)
-		if err != nil {
-			log.Errorf("node=%s, failed to generate config: %v", s.Cfg.ShortName, err)
-		}
+	err = s.GenerateConfig(cfgPath, cfgTemplate)
+	if err != nil {
+		log.Errorf("node=%s, failed to generate config: %v", s.Cfg.ShortName, err)
```
Comment (hellt) [inline nodes/srl/srl.go:493]: @steiler this piece was needed I think.
without it. gen config func was never called when startup config was empty, hence no log message was printed

Comment (steiler) [reply nodes/srl/srl.go:493]: Alright, then I think it is fine.

**Pull request merged**